aviatrix/resource_aviatrix_transit_gateway.go

```diff
@@ -1909,6 +1941,21 @@ func resourceAviatrixTransitGatewayRead(d *schema.ResourceData, meta interface{}
 				return fmt.Errorf("could not set interfaces into state: %v", err)
 			}
 		}
+		// set interface mapping
+		if len(gw.InterfaceMapping) != 0 {
+			var interfaceMapping []map[string]interface{}
+			sortedInterfaceMappings := sortInterfaceMappingByCustomOrder(gw.InterfaceMapping)
```

Comment:
Can you add a comment with an explanation of why the special sorting is needed? Future maintainers will thank you :)